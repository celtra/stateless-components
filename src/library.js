import '@/stateless/define_helpers'

import * as filtersObject from '@/helpers/filters.js'
import * as directivesObject from '@/helpers/directives.js'
import * as itemsUtilsObject from '@/stateless/items_utils.js'

import CheckboxComponent from '@/stateless/checkbox.vue'
import DialogButtonComponent from '@/stateless/dialog_button.vue'
import DialogComponent from '@/stateless/dialog.vue'
import DropAreaComponent from '@/stateless/drop_area.vue'
import FileUploadRequirementsComponent from '@/stateless/file_upload_requirements.vue'
import FileUploadComponent from '@/stateless/file_upload.vue'
import ImageListComponent from '@/stateless/image_list.vue'
import InputComponent from '@/stateless/input.vue'
import MultiselectComponent from '@/stateless/multiselect.vue'
import RadioButtonComponent from '@/stateless/radiobutton.vue'
import SelectboxComponent from '@/stateless/selectbox.vue'
import SupportTextComponent from '@/stateless/support_text.vue'
import IconComponent from '@/stateless/icon.vue'
import ToastComponent from '@/stateless/Toast.vue'
import PieChartComponent from '@/stateless/PieChart.vue'
import InlineDialogComponent from '@/stateless/InlineDialog.vue'
import DefaultListComponent from '@/stateless/DefaultList.vue'
import DefaultListItemComponent from '@/stateless/DefaultListItem.vue'
import ScrollableListComponent from '@/stateless/ScrollableList.vue'
import TypeaheadComponent from '@/stateless/Typeahead.vue'
import TypeaheadMultiselectComponent from '@/stateless/TypeaheadMultiselect.vue'
import TooltipComponent from '@/stateless/Tooltip.vue'

export const filters = filtersObject
export const directives = directivesObject
export const itemsUtils = itemsUtilsObject

export const Checkbox = CheckboxComponent
export const DialogButton = DialogButtonComponent
export const Dialog = DialogComponent
export const DropArea = DropAreaComponent
export const FileUploadRequirements = FileUploadRequirementsComponent
export const FileUpload = FileUploadComponent
export const ImageList = ImageListComponent
export const Input = InputComponent
export const Multiselect = MultiselectComponent
export const RadioButton = RadioButtonComponent
export const Selectbox = SelectboxComponent
export const SupportText = SupportTextComponent
export const Icon = IconComponent
export const Toast = ToastComponent
export const PieChart = PieChartComponent
export const InlineDialog = InlineDialogComponent
export const DefaultList = DefaultListComponent
export const DefaultListItem = DefaultListItemComponent
export const ScrollableList = ScrollableListComponent
export const Typeahead = TypeaheadComponent
export const TypeaheadMultiselect = TypeaheadMultiselectComponent
export const Tooltip = TooltipComponent
