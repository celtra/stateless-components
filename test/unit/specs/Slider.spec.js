import { Slider } from '@/library.js'
import Vue from 'vue'
import snapshotMixin from '../component_snapshot_mixin'

const Constructor = Vue.extend(Slider)
let vm = null
const dragEvent = { clientX: 300, preventDefault: () => {} }

describe('Slider', () => {
    beforeEach(() => {
        vm = new Constructor({
            propsData: {
                min: 0.1,
                max: 10,
                step: 0.1,
                theme: 'light',
                size: 'normal',
                label: 'Basic slider',
                value: 5,
                unit: '%',
                alignment: 'right',
            },
        }).$mount()

        vm.isDomReady = false
        vm.$refs.bar = { clientWidth: 300, getBoundingClientRect: () => {
            return { x: 160 }
        }, focus: () => {} }
    })

    snapshotMixin(Slider)

    describe('computed', () => {
        describe('limitValue', () => {
            it('should be max', () => {
                expect(vm.limitValue).toBe(10)
            })

            it('should be limit', () => {
                vm.limit = 8

                expect(vm.limitValue).toBe(8)
            })
        })

        describe('minLabelValue', () => {
            it('should be min', () => {
                expect(vm.minLabelValue).toBe('0.1')
            })

            it('should be minLabel', () => {
                vm.minLabel = 'minLabel'

                expect(vm.minLabelValue).toBe('minLabel')
            })
        })

        describe('maxLabelValue', () => {
            it('should be limitValue', () => {
                expect(vm.maxLabelValue).toBe('10')
            })

            it('should be maxLabel', () => {
                vm.maxLabel = 'maxLabel'

                expect(vm.maxLabelValue).toBe('maxLabel')
            })
        })

        describe('stepsCount', () => {
            it('should be 99', () => {
                expect(vm.stepsCount).toBe(99)
            })
        })

        describe('position', () => {
            it('should be 0', () => {
                expect(vm.position).toBe(0)
            })

            it('should not be close to 148.48', () => {
                vm.isDomReady = true

                expect(vm.position).toBeCloseTo(148.48)
            })
        })

        describe('sliderWidth', () => {
            it('should be zero', () => {
                expect(vm.sliderWidth).toBe(0)
            })

            it('should be some positive number', () => {
                vm.isDomReady = true

                expect(vm.sliderWidth).toBe(300)
            })
        })

        describe('decimalPlacesCount', () => {
            it('should be 1', () => {
                expect(vm.decimalPlacesCount).toBeCloseTo(1)
            })

            it('should be 0', () => {
                vm.min = 0
                vm.step = 1

                expect(vm.decimalPlacesCount).toBe(0)
            })
        })

        describe('isWholeNumber', () => {
            it('should be false', () => {
                expect(vm.isWholeNumber).toBe(false)
            })

            it('should be true', () => {
                vm.min = 0
                vm.step = 1

                expect(vm.isWholeNumber).toBe(true)
            })
        })

        describe('lastActiveIndex', () => {
            it('should be close to 9.4', () => {
                expect(vm.lastActiveIndex).toBeCloseTo(9.4)
            })
        })

        describe('labelsActiveClass', () => {
            it('should be min === true, max === false', () => {
                expect(vm.labelsActiveClass).toEqual({
                    min: { active: true },
                    max: { active: false },
                })
            })
        })
    })

    describe('methods', () => {
        describe('startDrag', () => {
            it('should return undefined', () => {
                vm.disabled = true

                expect(vm.startDrag(dragEvent)).toBeUndefined()
            })

            it('should set isChanged and isDragging to true', () => {
                spyOn(vm, 'setPosition')

                vm.startDrag(dragEvent)
                expect(vm.isChanged).toBe(true)
                expect(vm.isDragging).toBe(true)
                expect(vm.setPosition).toHaveBeenCalled()
            })
        })

        describe('setPosition', () => {
            it('should emit new value', () => {
                spyOn(vm, '$emit')

                vm.setPosition(dragEvent)
                expect(vm.$emit).toHaveBeenCalledWith('input', 10)
            })
        })

        describe('stopDrag', () => {
            it('should set isDragging to false', () => {
                vm.stopDrag()
                expect(vm.isDragging).toBe(false)
            })
        })

        describe('handleStepDecrease', () => {
            it('should emit decreased value', () => {
                spyOn(vm, '$emit')
                vm.handleStepDecrease()
                expect(vm.$emit).toHaveBeenCalled()
            })

            it ('should not emit anything', () => {
                vm.value = 0.1

                spyOn(vm, '$emit')
                vm.handleStepDecrease()
                expect(vm.$emit).not.toHaveBeenCalled()
            })
        })

        describe('handleStepIncrease', () => {
            it('should emit increased value', () => {
                spyOn(vm, '$emit')
                vm.handleStepIncrease()
                expect(vm.$emit).toHaveBeenCalled()
            })

            it ('should not emit anything', () => {
                vm.value = 100

                spyOn(vm, '$emit')
                vm.handleStepIncrease()
                expect(vm.$emit).not.toHaveBeenCalled()
            })
        })

        describe('handleInput', () => {
            it('should proxy event', () => {
                spyOn(vm, '$emit')
                vm.handleInput('value')
                expect(vm.$emit).toHaveBeenCalledWith('input', 'value')
            })
        })

        describe('isValidInput', () => {
            it('should be null', () => {
                expect(vm.isValidInput(5)).toBeNull()
            })

            it('should be empty string', () => {
                expect(vm.isValidInput(0)).toBe('')
            })

            it('should be empty string', () => {
                expect(vm.isValidInput(101)).toBe('')
            })
        })

        describe('tickClass', () => {
            it('should be active and hidden', () => {
                vm.isDomReady = true

                expect(vm.tickClass(1)).toEqual({
                    'active': true,
                    'hidden': true,
                })
            })

            it('should be inactive and shown', () => {
                vm.isDomReady = true

                expect(vm.tickClass(15)).toEqual({
                    'active': false,
                    'hidden': false,
                })
            })
        })
    })
})
