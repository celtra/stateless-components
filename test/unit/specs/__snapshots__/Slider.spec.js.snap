// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match the snapshot 1`] = `
<div
  class="slider slider--light slider--normal"
>
  <div
    class="slider__row"
  >
    <div
      class="slider__label"
    >
      
            Basic slider
        
    </div>
  </div>
   
  <div
    class="slider__row"
  >
    <div
      class="slider__input"
    >
      <div
        class="input input--normal input--light"
      >
        <!---->
         
        <div
          class="input-field"
          title=""
        >
          <!---->
           
          <div
            class="input-row"
          >
            <!---->
             
            <div
              class="input-row__input-flex"
            >
              <!---->
               
              <!---->
               
              <input
                class="input-row__placeholder-text"
                placeholder="5.0"
                style="text-align: right;"
                type="text"
              />
            </div>
             
            <div
              class="input-row__unit input-row__unit--right"
            >
              <span>
                %
              </span>
            </div>
          </div>
           
          <div
            class="input-field__border-overlay"
          />
           
          <div
            class="input-field__message-wrap"
          >
            <div
              class="input-field__helper-text"
            >
              
                    
                
            </div>
             
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="slider-bar"
      tabindex="0"
    >
      <div
        class="slider-bar__container"
      >
        <div
          class="slider-ruler"
        >
          <div
            class="slider-ruler__label slider-ruler__label--active"
          >
            0.1
          </div>
           
          <div
            class="slider-ruler__ticks"
          >
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick slider-ruler__tick--active"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
            <div
              class="slider-ruler__tick"
            />
          </div>
           
          <div
            class="slider-ruler__label"
          >
            10
          </div>
        </div>
         
        <div
          class="slider-bar__rail"
        >
          <div
            class="slider-rail slider-rail--passive"
          />
           
          <div
            class="slider-rail slider-rail--active"
            style="width: 0px;"
          />
           
          <div
            class="slider-rail__handle"
            style="left: 0px;"
          >
            <div
              class="slider-rail__handle-dot"
            />
          </div>
        </div>
      </div>
       
      <div
        class="slider-helper-text"
      />
    </div>
  </div>
</div>
`;
